analyzers/tests/SonarAnalyzer.UnitTest/Json/JsonWalkerTest.cs

```diff
@@ -0,0 +1,65 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.Json;
+
+namespace SonarAnalyzer.UnitTest.Common
+{
+    [TestClass]
+    public class JsonWalkerTest
+    {
+        [TestMethod]
+        public void VisitsAllNodes()
```

Comment:
Add a parameterized test with empty json:
* `[]`
* `{}`
* `null`
* `""`
* `0`
* `true`
* `false`

Reply:
Atomic values are not supported for parsing, I've added only `[]` and `{}`

Reply:
> Atomic values are not supported for parsing

Those are valid json but as long as we don't have a problem with it, that's fine. Is it worth fixing it?

Reply:
No. There's no point in analyzing `true` or `0` file during analysis, there's nothing to learn about the project configuration there